### DISCUSSION TITLE
What is the cost of creating and setting a new default subscriber?

### DISCUSSION THREAD

**Discussion by rimutaka** [tokio-rs/tracing#3029] [Q&A]
What is the cost of running these 2 lines of code?
```rust
    let sub = tracing_subscriber::fmt().without_time().finish();
    let _guard = tracing::subscriber::set_default(sub);
```

I looks like line 1 involved creating structs with default values, an Arc and an RW Lock and should be lightning fast.

The code behind line 2 was getting a bit too complicated for me to estimate.
Looks like it depends on the number of the callsites and the size of the registry.

Contrary to my research, I was getting about 10us duration for line 1 and 0 for line 2 in my minimal test.

I'll appreciated any insight you can share.

**Comment by davidbarsky** [comment]
> I looks like line 1 involved creating structs with default values, an Arc and an RW Lock and should be lightning fast.

Yes, that is correct. It's not actually _doing_ anything beyond 

> The code behind line 2 was getting a bit too complicated for me to estimate.

It is `O(number_of_spans_and_events)`[^1], but that is also the normal work that `tracing` does when a subscriber first encounters spans and events. I feel like there's a question behind this question: what's your motivation in asking this?

[^1]: It also skips past anything that is below the activated level.

**Comment by rimutaka** [comment]
@davidbarsky , thank you for the explanation.
I was trying to remove the output of a span added by a previous service in the chain.
One solution was to use its own subscriber inside the inner service, but it may be a bad idea if the service is called in a loop.
